src/Utilities/src/RootSolvers.jl

```diff
@@ -2,162 +2,97 @@
+Finds the nearest root of `f` to `x0` and `x1`. Returns a the value of the root `x` such
+that `f(x) ≈ 0`, and a Boolean value `converged` indicating convergence.
+
+`method` can be one of:
+- `SecantMethod()`: [Secant method](https://en.wikipedia.org/wiki/Secant_method)
+- `RegulaFalsiMethod()`: [Regula Falsi method](https://en.wikipedia.org/wiki/False_position_method#The_regula_falsi_(false_position)_method).
+
+The keyword arguments:
+- `xatol` and `xrtol` are the absolute and relative tolerances of the input.
+- `yatol` is the absolute tolerance of the output of `f`.
+- `maxiters` is the maximum number of iterations.
 
-Solves the root equation, `f`, using Secant method.
-See [RootSolvers](@ref) for more information.
 """
-function find_zero(f::F,
-                   x_0::T,
-                   x_1::T,
-                   args::Tuple,
-                   iter_params::IterParams{R, Int},
-                   method::SecantMethod
-                   )::Tuple{T, Bool} where {F, R, T <: Union{R, AbstractArray{R}}}
-  iter_max = iter_params.iter_max
-  tol_abs = iter_params.tol_abs
-  Δx = x_1 - x_0
-  y_0 = f(x_0, args...)
-  y_1 = f(x_1, args...)
-  x = x_1 - y_1 * Δx / (y_1 - y_0) # puts x in outer scope, & saves iter_max = 0 case
-  for i in 1:iter_max
-    x = x_1 - y_1 * Δx / (y_1 - y_0)
-    y_0 = y_1
-    x_0 = x_1
-    x_1 = x
-    y_1 = f(x_1, args...)
-    Δx = x_1 - x_0
-    if abs(y_1) < tol_abs
-      return x, true
+function find_zero end
+
+
+function find_zero(f::F, x0::T, x1::T, ::SecantMethod;
+                   xatol=zero(x0), xrtol=sqrt(eps(T)), yatol=sqrt(eps(T)), maxiters=10_000) where {F, T<:AbstractFloat}
+  y0 = f(x0)
+  y1 = f(x1)
+  for i in 1:maxiters
+    Δx = x1 - x0
+    Δy = y1 - y0
+    x0, y0 = x1, y1
+    x1 -= y1 * Δx / Δy
+    y1 = f(x1)
+    if isapprox(y1,0; atol=yatol) || isapprox(x0,x1; atol=xatol, rtol=xrtol)
+      return x1, true
     end
   end
-  return x, false
+  return x1, false
 end
 
-"""
-`find_zero(f, x_0, x_1, args, iter_params, RegulaFalsiMethod)`
-
-Solves the root equation, `f`, using Regula Falsi method.
-See [RootSolvers](@ref) for more information.
-"""
-function find_zero(f::F,
-                   x_0::T,
-                   x_1::T,
-                   args::Tuple,
-                   iter_params::IterParams{R, Int},
-                   method::RegulaFalsiMethod
-                   )::Tuple{T, Bool} where {F, R, T <: Union{R, AbstractArray{R}}}
-  iter_max = iter_params.iter_max
-  tol_abs = iter_params.tol_abs
-  a = x_0
-  b = x_1
-  fa = f(a, args...)
-  fb = f(b, args...)
-  c = a
-  i = 0
-  for i in 1:iter_max
-    fa = f(a, args...)
-    fb = f(b, args...)
-    c = (a * fb - b * fa)/ (fb - fa)
-    fc = f(c, args...)
-    if abs(fc) < tol_abs
-      return c, true
-    elseif fc * fa < 0
-      b = c
+function find_zero(f::F, x0::T, x1::T, ::RegulaFalsiMethod;
+                   xatol=zero(x0), xrtol=sqrt(eps(T)), yatol=sqrt(eps(T)), maxiters=10_000) where {F, T<:AbstractFloat}
```

Comment:
Should `xatol=zero(x0)` be `xatol=sqrt(eps(T))`?

Reply:
Relative tolerance by default tends to make more sense for this: these are the same defaults for [`isapprox`](https://docs.julialang.org/en/v1/base/math/#Base.isapprox).